File: animal.js
var Animal = function (name, type, sound) {  // this could also be `function Animal (name, type, sound) { ` 
  this.name = name;
  this.type = type;
  this.sound = sound;
};

Animal.prototype = {
  makeNoise: function () {
    return this.sound + " is the sound that " + this.name + " makes!";
  },
  getWeightAtAge: function(age) {
    if (this.type === "dog"){
      return 5 * age;
    } else if (this.type == "fish"){
      return 2 * age;
    } else {
      return age;
    }
  },
  toString: function () {
    return this.name + " is a " + this.type;
  }
  
}


var ziggy = new Animal("Ziggy", "dog", "Woof!");
var nemo = new Animal("Nemo", "fish", "bubble!");

print(ziggy.name + " is a " + ziggy.type + " and makes the sound " + ziggy.sound);

print(ziggy.makeNoise());
print(nemo.makeNoise());

print(nemo.toString());

print("When " + ziggy.name + " is 7 she'll be " + ziggy.getWeightAtAge(7) + " pounds!");

print("Setting Nemo's type to shark...");
nemo.type = "shark";
print("Now " + nemo.name + " is a " + nemo.type);

File: animal.rb
class Animal 
  attr_accessor :name, :type, :sound

  def initialize(name, type, sound)
    self.name = name
    self.type = type
    self.sound = sound
  end 

  def make_noise
    "#{self.sound} is the sound that #{self.name} makes."
  end

  def get_weight_at_age(age)
    if self.type == "dog"
      5 * age
    elsif self.type == "fish"
      2 * age
    else 
      age
    end
  end

  def to_s
    "#{self.name} is a #{self.type}"
  end 

end 

ziggy = Animal.new("Ziggy", "dog", "Woof!")
nemo = Animal.new("Nemo", "fish", "bubble!")

puts ziggy.name + " is a " + ziggy.type + " and makes the sound " + ziggy.sound
puts ziggy.make_noise
puts nemo.make_noise

puts nemo.to_s

puts "When " + ziggy.name + " is 7 she'll be " + ziggy.get_weight_at_age(7).to_s + " pounds!"

puts "Setting Nemo's type to shark..."
nemo.type = "shark";
puts "Now " + nemo.name + " is a " + nemo.type
